nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/test/java/org/apache/nifi/processors/azure/storage/ITListAzureBlobStorage.java

```diff
@@ -55,12 +55,58 @@ public void testListBlobsUsingCredentialService() throws Exception {
         assertResult();
     }
 
+    @Test
+    public void testListWithMinAge() throws Exception {
+        runner.setProperty(ListAzureBlobStorage.MIN_AGE, "1 hour");
+
+        runner.assertValid();
+        runner.run(1);
+
+        runner.assertTransferCount(ListAzureBlobStorage.REL_SUCCESS, 0);
+    }
+
+    @Test
+    public void testListWithMaxAge() throws Exception {
+        runner.setProperty(ListAzureBlobStorage.MAX_AGE, "1 hour");
+
+        runner.assertValid();
+        runner.run(1);
+
+        assertResult(TEST_FILE_CONTENT);
+    }
+
+    @Test
+    public void testListWithMinSize() throws Exception {
+        uploadTestBlob("nifi-test-blob2", "Test");
+        runner.setProperty(ListAzureBlobStorage.MIN_SIZE, "5 B");
+
+        runner.assertValid();
+        runner.run(1);
+
+        assertResult(TEST_FILE_CONTENT);
+    }
+
+    @Test
+    public void testListWithMaxSize() throws Exception {
+        uploadTestBlob("nifi-test-blob2", "Test");
```

Comment:
This test fails for me when I run it on Java 8. It seems the "lag" between uploading and listing would need to be applied here too as in the `setUp()` method.
Interestingly, it works on Java 11.

The same is true for `testListWithMinSize()`. It still succeeds even if I modify the test data to larger than the min size.